File: src/minigames/eggventure/src/main/java/work/lclpnet/ap2/game/eggventure/EggventureMiniGame.java
package work.lclpnet.ap2.game.eggventure;

import net.minecraft.core.RegistryAccess;
import net.minecraft.resources.Identifier;
import net.minecraft.world.item.ItemStack;
import org.jetbrains.annotations.NotNull;
import work.lclpnet.ap2.ApConstants;
import work.lclpnet.ap2.api.game.*;
import work.lclpnet.ap2.impl.util.ApRegistries;
import work.lclpnet.ap2.impl.util.heads.PlayerHeads;

public class EggventureMiniGame implements MiniGame {

    @Override
    public boolean canBeFinale(@NotNull GameStartContext context) {
        return true;
    }

    @Override
    public boolean canBePlayed(@NotNull GameStartContext context) {
        return true;
    }

    @Override
    public @NotNull MiniGameInstance createInstance(@NotNull MiniGameHandle gameHandle) {
        return new EggventureInstance(gameHandle);
    }

    @Override
    public @NotNull Identifier getId() {
        return ApConstants.identifier("eggventure");
    }

    @Override
    public @NotNull GameType getType() {
        return GameType.FFA;
    }

    @Override
    public @NotNull String getAuthor() {
        return ApConstants.PERSON_LCLP;
    }

    @Override
    public @NotNull ItemStack getIcon(@NotNull RegistryAccess manager) {
        return manager.lookupOrThrow(ApRegistries.PLAYER_HEAD)
                .getValueOrThrow(PlayerHeads.EASTER_EGG_PINK_PATTERN)
                .createStack();
    }
}

File: src/minigames/eggventure/src/main/kotlin/work/lclpnet/ap2/game/eggventure/EggventureMiniGame.kt
package work.lclpnet.ap2.game.eggventure

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.base.GameStartContext
import work.lclpnet.ap2.game.GameType
import work.lclpnet.ap2.game.MiniGame
import work.lclpnet.ap2.game.MiniGameFactory
import work.lclpnet.ap2.impl.util.ApRegistries
import work.lclpnet.ap2.impl.util.heads.PlayerHeads

class EggventureMiniGame : MiniGame {
    override val id = ApConstants.identifier("eggventure")
    override val type = GameType.FFA
    override val author = ApConstants.PERSON_LCLP
    override fun canBeFinale(context: GameStartContext): Boolean = true
    override fun canBePlayed(context: GameStartContext): Boolean = true
    override fun createFactory(): MiniGameFactory = EggventureFactory()
    override fun getIcon(manager: RegistryAccess): ItemStack = manager.lookupOrThrow(ApRegistries.PLAYER_HEAD)
        .getValueOrThrow(PlayerHeads.EASTER_EGG_PINK_PATTERN)
        .createStack()
}
